docs/remote/devcontainer-cli.md

```diff
@@ -126,13 +126,96 @@ These steps above are also provided in the CLI repo's [README](https://github.co
+
+### Example of building and publishing an image
+
+For example, you may want to pre-build a number of images that you then reuse across multiple projects or repositories. To do so, follow these steps:
+
+1. [Create](/docs/editor/versioncontrol.md#initialize-a-repository) a source code repository.
+
+1. Create dev container configuration for each image you want to pre-build, customizing as you wish (including [dev container Features](#dev-container-features-preview)). For example, consider this devcontainer.json file:
+
+    ```json
+    {
+        "build": {
+            "dockerfile": "Dockerfile"
+        },
+        "features": {
+            "ghcr.io/devcontainers/features/docker-in-docker:1": {
+            "version": "latest"
+            }
+        }
+    }
+    ```
+
+1. Use the `devcontainer build` command to build the image and [push](https://docs.docker.com/engine/reference/commandline/push/) it to your image registry. See documentation for your image registry (like the [Azure Container Registry](https://docs.microsoft.com/azure/container-registry/container-registry-get-started-docker-cli?tabs=azure-cli), [GitHub Container Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#pushing-container-images), or [Docker Hub](https://docs.docker.com/engine/reference/commandline/push)) for information on image naming and additional steps like authentication.
+
+    ```bash
+    devcontainer build --workspace-folder <my_repo> --push true --image-name <my_image_name>:<optional_image_version>
+    ```
+
+1. Create a simplified devcontainer.json in repositories where you'd like to use the image - the devcontainer.json should either use the `image` property or reference the image in an associated Docker Compose file. Include any dev container Features you added in your pre-build configuration. For example:
+
+        ```json
+        {
+            "image": "ghcr.io/your-org/your-image-name",
+            "features": {
+                "ghcr.io/devcontainers/features/docker-in-docker:1": {
+                "version": "latest"
+                 }
+            }
+        }
+        ```
+
+On the other hand, if you only intend to use the pre-built image from one repository, you can use the `cacheFrom` property in devcontainer.json or `cache_from` in a related Docker Compose file instead. This will download the image and treat its image layers like a local cache even if this is the first time you've created the Dockerfile on your machine. Like the option above, be sure to include any dev container Features. For example:
+
+```json
+{
+    "build": {
+        "dockerfile": "Dockerfile",
+        "cacheFrom": "ghcr.io/your-org/your-image-name"
+    },
+    "features": {
+        "ghcr.io/devcontainers/features/docker-in-docker:1": {}
+    }
+}
+
+### [Optional] Avoiding problems with images built using Docker
```

Comment:
Is this too much detail maybe? (Also: Would we want to avoid "vscode" and use "devcontainer" instead?)

Reply:
Yeah, upon a re-read, it does feel a bit specific and VS Code-centric. We had added it based on previous user request, but I think that may have been a singular or small number of users, so I don't anticipate this would be a big point of confusion. Removed, cc @Chuxel if you have other thoughts.

Reply:
This bit of doc was actually contributed by a user who apparently hit this, so not sure we want to remove it, but it's more of a "tip" than a requirement - it could be moved to an "Advanced" topic (though maybe linked from here)?  Agree that using `devcontainer-cli` as the ARG name is better than vscode tho.

Reply:
It might be loosely enough related to https://code.visualstudio.com/remote/advancedcontainers/reduce-docker-warnings, that I could add the info there?

Reply:
I just pushed a change to add this info in https://code.visualstudio.com/remote/advancedcontainers/reduce-docker-warnings and to point to it from devcontainer-cli.md. I also updated the ARG name to `devcontainercli`.